index.d.ts

```diff
@@ -258,8 +258,8 @@ export interface CollectionWhereOperations {
 export type CollectionWhereCallback<
   E extends { [name: string]: Entity<any, any, any, any> },
   I extends Partial<AllEntityAttributes<E>>,
-> = <W extends { [A in keyof I]: WhereAttributeSymbol<I[A]> }>(
-  attributes: W,
+> = (
+  attributes: { [A in keyof I]: WhereAttributeSymbol<I[A]> },
```

Comment:
@tywalch I'm not sure if there is a specific reason for this to be generic. 
I couldn't get TypeScript to allow the following type:

```tsx
Pick<
  I,
  CollectionProjectedAttributeNames<E, Collections, Collection>
>
```

to satisfy `Partial<AllEntityAttributes<E>>`. 

But after removing the generic in the `where` callback, it worked. 
let me know if the generic addition is intentional.